pkgs/development/tools/java/jprofiler/default.nix

```diff
@@ -51,49 +51,32 @@ let
     categories = [ "Development" ];
   };
 
-  linux = stdenv.mkDerivation {
-    inherit pname version src desktopItems;
-
-    nativeBuildInputs = [ makeWrapper copyDesktopItems ];
+  dontStrip = true;
 
-    installPhase = ''
-      runHook preInstall
-      cp -r . $out
+  installPhase = ''
+    runHook preInstall
 
-      rm -f $out/bin/updater
-      rm -rf $out/bin/linux-ppc*
-      rm -rf $out/bin/linux-armhf
-      rm -rf $out/bin/linux-musl*
+    cp -r . $out
 
-      for f in $(find $out/bin -type f -executable); do
-        wrapProgram $f --set JAVA_HOME "${jdk.home}"
-      done
+    for f in $(find $out/bin $out/.install4j -type f -executable); do
+      wrapProgram $f --set JAVA_HOME "${jdk.home}"
+    done
 
-      install -Dm644 "${srcIcon}" \
+    ${lib.optionalString (component == "ui") ''
+      install -Dm644 "${fetchurl artefacts.icon}" \
         "$out/share/icons/hicolor/scalable/apps/jprofiler.png"
-      runHook postInstall
-    '';
-
-    meta = meta // { platforms = lib.platforms.linux; };
-  };
 
-  darwin = stdenv.mkDerivation {
-    inherit pname version src;
+      ${lib.optionalString stdenv.isDarwin ''
+        mkdir -p $out/Applications
+        ln -s $out/bin/${pname}.app $out/Applications/${nameApp}.app
+        ln -s ${jdk.home} $out/.install4j/jre
+      ''}
+    ''}
```

Comment:
Please use ```lib.optionalString bool ''
  ...
``
```.

Reply:
@SuperSandro2000 I don't get your comment. Sorry, can you rephrase it?

Reply:
something with formatting got messed up but essentially we should optionally append a multiline string instead of nesting it into variable.